### PR TITLE
Support for optional explicit BinarySuffix precision

### PR DESCRIPTION
**Pull request by mostertb**
When humanizing a number of bytes into the largest possible unit (as the BinarySuffix humanizer already does) you may have differing requirements in the number of decimal places you wish to show. 

For example, you may which to truncate the result to a natural number if displaying the value within a paragraph or you may wish to show three decimal places if you want to still maintain information about the next largest unit.

My change gives this functionality via an optional third precision parameter to Number::binarySuffix(). 
The parameter being optional means that the change is non-breaking.

I have attempted to make the change in such as way that it will also be compatible with future updates to the existing hard-coded ICU 56.1 decimal formats used by BinarySuffix

Lastly, I have also updated the README to include this new functionality and provided a similar test coverage to the existing functionality.

### CODE REVIEW

**Comment by norzechowicz** [comment]
Yea that is definitely something that we need, I would just like to make ``$precision``, second, not third parameter, but this would be a BC Break. What would you say about adding new method to Number
``Number::preciseBinarySuffix($number, $precision, $locale = 'en');`` where $precision is not optional?

**Comment by mostertb** [comment]
Thank you for the feedback. I'm glad that I could add something useful.

I happy to add a `PHP Number::preciseBinarySuffix(...)` function. I'll update the pull request later today

**Comment by mostertb** [comment]
I've added `PHP Number::preciseBinarySuffix(...)` 
Also updated the README and moved the tests in to phpunit

File: tests/Coduo/PHPHumanizer/Tests/NumberTest.php [hunk 1]
```diff
@@ -166,6 +179,28 @@ public function binarySuffixDataProvider()
...
+            array(-1, -1, 3),
+            array("0 bytes", 0, 3),
+            array("1 bytes", 1, 3),
+            array("1.000 kB", 1024, 3),
+            array("1.001 kB", 1025, 3),
+            array("1.500 kB", 1536, 3),
+            array("5.000 MB", 1048576 * 5, 3),
+            array("2.000 GB", 1073741824 * 2, 3),
+            array("3.000 TB", 1099511627776 * 3, 3),
+            array("1.178 PB", 1325899906842624, 3),
```
Comment (norzechowicz) [inline tests/Coduo/PHPHumanizer/Tests/NumberTest.php]: Could you please add some tests for precision that is bigger and lower than 3? For example, 1,2,3,4,5,6,7  ?

**Comment by mostertb** [comment]
Hi Norbert 

I've improved the test as requested and added cases for exceptions. It was a worthwhile exercise as I did find an issue that has been addressed in 
https://github.com/mostertb/php-humanizer/commit/27a65adb9be7ad787a26c218cc3fe4019d90f183

**Comment by norzechowicz** [comment]
@mostertb awesome, good job! Thanks a lot :beers:

**Pull request merged**